### DISCUSSION TITLE
Convert plantuml state diagram to xstate State diagram data for rendering

### DISCUSSION THREAD

**Discussion by lei4519** [statelyai/xstate#4400] [Ideas]
I want to input (import) PlantUML state chart data into xstate-viz to render xstate state charts. I'm currently working on implementing a PlantUML state chart parser for this conversion. Do you have any suggestions or plans regarding this?

**Comment by davidkpiano** [comment]
I think this is a great idea. Keep in mind that there are many statechart features that PlantUML does not natively offer, and there are a couple things that may not easily translate one-to-one with PlantUML, but for most use-cases, it should be straightforward to convert between the two.

**Reply by lei4519** [reply]
Thanks. I'll continue my work
